tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -251,11 +251,14 @@ public void onFailure(Throwable e) {
             private void doRequestEndProcessing() {
                 if (span != null) {
                     span.finish();
+                    TracingContext tracingContext = topChannel.getTracingContext();
```

Comment:
that's the point of using instrumented executor that propagates the tracing context, isn't it?
of course, it's set up outside of this code, so responsibility is on those who want to use specific tracing implementation to properly instrument the execution environment.

`.sendRequestToAsyncHandler()` that initiates the tracing is executed on the same thread as `.handleAsync()` does, so context gets full chance to be properly propagated.
the thread that resolves the response future should also be instrumented to have the context, so propagation is covered on the way back, too.